packages/core/src/new-api/internal/deployer.ts

```diff
@@ -113,6 +115,10 @@ export class Deployer {
     const batches = Batcher.batch(module, previousStateMap);
 
     return this._executionEngine.execute({
+      block: { number: -1, hash: "-" },
```

Comment:
I commented this on PR B, but I think everything that's passed here should be the internal state of the ExecutionEngine, and it should receive the necessary params instead.

Reply:
Agreed, I have recorded a todo for this: https://github.com/NomicFoundation/ignition/issues/311